### PR TITLE
NNS fnd

### PR DESCRIPTION
**Pull request by PikalaxALT**

### CODE REVIEW

File: arm9/lib/libnns/src/NNS_FND_allocator.c [hunk 1]
```diff
@@ -0,0 +1,32 @@
+#include "nitro.h"
+#include "NNS_FND_allocator.h"
+#include "NNS_FND_expheap.h"
+
+void *AllocatorAllocForExpHeap(NNSFndAllocator * pAllocator, u32 size)
```
Comment (red031000) [inline arm9/lib/libnns/src/NNS_FND_allocator.c:5]: these funcs need ARM_FUNC

File: arm9/lib/libnns/src/NNS_FND_expheap.c [hunk 2]
```diff
@@ -0,0 +1,385 @@
...
+{
+    return GetExpHeapHeadPtrFromHeapHead(heap);
+}
+
+static inline NNSiFndHeapHead* GetHeapHeadPtrFromExpHeapHead(NNSiFndExpHeapHead* pEHHead)
+{
+    return SubU32ToPtr(pEHHead, sizeof(NNSiFndHeapHead));
+}
+
+void GetRegionOfMBlock(NNSiMemRegion* region, NNSiFndExpHeapMBlockHead* block)
```
Comment (red031000) [inline arm9/lib/libnns/src/NNS_FND_expheap.c:69]: ARM_FUNC

File: arm9/lib/libnns/src/NNS_FND_expheap.c [hunk 3]
```diff
@@ -0,0 +1,385 @@
...
+        size = 1;
+    size = NNSi_FndRoundUp(size, 4);
+    if (alignment >= 0)
+        return AllocFromHead(handle, size, alignment);
+    else
+        return AllocFromTail(handle, size, -alignment);
+}
+
+/*
+u32 NNS_FndResizeForMBlockExpHeap(NNSFndHeapHandle heap, void *memBlock, u32 size)
```
Comment (red031000) [inline arm9/lib/libnns/src/NNS_FND_expheap.c:337]: commented out? does this not match?

File: arm9/lib/libnns/src/NNS_FND_heapcommon.c [hunk 4]
```diff
@@ -0,0 +1,55 @@
+#include "nitro.h"
+#include "NNS_FND_heapcommon.h"
+#include "NNS_FND_list.h"
+
+BOOL sRootListInitialized;
+NNSFndList sRootList;
+
+void *NNS_FndGetNextListObject(NNSFndList *, void *);
+
+static NNSiFndHeapHead* FindContainHeap(NNSFndList * pList, const void * memBlock)
```
Comment (red031000) [inline arm9/lib/libnns/src/NNS_FND_heapcommon.c:10]: ARM_FUNC

File: arm9/lib/libnns/src/NNS_FND_list.c [hunk 5]
```diff
@@ -0,0 +1,128 @@
+#include "nitro.h"
+#include "NNS_FND_list.h"
+
+#define OBJ_TO_LINK(list, obj) ((NNSFndLink*)((void*)(obj) + (list)->offset))
+
+void NNS_FndInitList(NNSFndList* list, s32 alignment)
```
Comment (red031000) [inline arm9/lib/libnns/src/NNS_FND_list.c:6]: ARM_FUNC

**Review by red031000 — CHANGES_REQUESTED** [review]
few small things

File: arm9/lib/libnns/src/NNS_FND_allocator.c (hunk 1 above)
Comment (PikalaxALT) [reply arm9/lib/libnns/src/NNS_FND_allocator.c:5]: i really want to move away from the ARM_FUNC/THUMB_FUNC standard in favor of just writing the thumb pragma in global.h

File: arm9/lib/libnns/src/NNS_FND_expheap.c (hunk 3 above)
Comment (PikalaxALT) [reply arm9/lib/libnns/src/NNS_FND_expheap.c:337]: yeah, i couldn't match it, that's part of the reason why i'm making this a stopping point. there's still the rest of expheap and all of frameheap left in NNS fnd

**Review by red031000 — APPROVED** [review]

**Pull request merged**